packages/multiple-did-identities/src/__tests__/generate-key-pair.spec.ts

```diff
@@ -46,3 +50,115 @@ describe('generateKeyPairs', () => {
+      kty: 'OKP',
+      crv: 'Ed25519',
+      x: base64UrlEncode(rawPublicKey),
+    });
+
+    // Validate privateKeyJwk
+    expect(privateKeyJwk).toMatchObject({
+      id: '#key-1',
+      type: 'JsonWebKey2020',
+      privateKeyJwk: {
+        kty: 'OKP',
+        crv: 'Ed25519',
+        d: base64UrlEncode(rawPrivateKey),
+        x: base64UrlEncode(rawPublicKey),
+      },
+    });
+  });
+
+  it('should produce distinct key pairs for each iteration', async () => {
+    const numKeys = 2;
+    const keys = await generateKeyPairsED25519(numKeys);
+
+    // Ensure each key pair is unique
+    const [firstKeyPair, secondKeyPair] = keys;
+
+    // Raw keys must be unique
+    expect(firstKeyPair.rawPublicKey).not.toEqual(secondKeyPair.rawPublicKey);
+    expect(firstKeyPair.rawPrivateKey).not.toEqual(secondKeyPair.rawPrivateKey);
+
+    // JWK representations must also be unique
+    expect(firstKeyPair.publicKeyJwk).not.toEqual(secondKeyPair.publicKeyJwk);
+    expect(firstKeyPair.privateKeyJwk).not.toEqual(secondKeyPair.privateKeyJwk);
+  });
+});
+
+describe('generateKeyPairsX25519', () => {
+  it('should generate the correct number of key pairs', async () => {
+    const numKeys = 2;
+    const keys = await generateKeyPairsX25519(numKeys);
+
+    // Verify the correct number of keys
+    expect(keys).toHaveLength(numKeys);
+  });
+
+  it('should generate valid key pairs with raw keys and JWK representations', async () => {
+    const numKeys = 1;
+    const keys = await generateKeyPairsX25519(numKeys);
+    const { rawPublicKey, rawPrivateKey, publicKeyJwk, privateKeyJwk } =
+      keys[0];
+
+    // Verify that raw keys are Uint8Arrays
+    expect(rawPublicKey).toBeInstanceOf(Uint8Array);
+    expect(rawPrivateKey).toBeInstanceOf(Uint8Array);
+
+    // Verify the structure of the publicKeyJwk
+    expect(publicKeyJwk).toMatchObject({
+      kty: 'OKP',
+      crv: 'X25519',
+      x: base64UrlEncode(rawPublicKey),
+    });
+
+    // Verify the structure of the privateKeyJwk
+    expect(privateKeyJwk).toMatchObject({
+      id: '#key-2',
+      type: 'JsonWebKey2020',
+      privateKeyJwk: {
+        kty: 'OKP',
+        crv: 'X25519',
+        d: base64UrlEncode(rawPrivateKey),
+        x: base64UrlEncode(rawPublicKey),
+      },
+    });
+  });
+
+  it('should produce distinct key pairs for each iteration', async () => {
+    const numKeys = 2;
+    const keys = await generateKeyPairsX25519(numKeys);
+
+    // Ensure distinct key pairs
+    const [firstKeyPair, secondKeyPair] = keys;
+
+    // Raw keys should be unique
+    expect(firstKeyPair.rawPublicKey).not.toEqual(secondKeyPair.rawPublicKey);
+    expect(firstKeyPair.rawPrivateKey).not.toEqual(secondKeyPair.rawPrivateKey);
+
+    // JWK representations should also be unique
+    expect(firstKeyPair.publicKeyJwk).not.toEqual(secondKeyPair.publicKeyJwk);
+    expect(firstKeyPair.privateKeyJwk).not.toEqual(secondKeyPair.privateKeyJwk);
+  });
+});
```

Comment:
The assertions for public/private keys and JWKs are repetitive across the tests. You could use a helper function to validate the structure of a key pair to improve readability, WDYT @forkimenjeckayang. ?